lib/url-pattern/url-match-pattern.js

```diff
@@ -51,7 +54,17 @@ _.assign(UrlMatchPattern.prototype, /** @lends UrlMatchPattern.prototype */ {
          * The url match pattern string
          * @type {String}
          */
-        this.pattern = _.get(options, 'pattern') || MATCH_ALL_URLS;
+        this.pattern = _.get(options, 'pattern') || this.pattern || MATCH_ALL_URLS;
```

Comment:
This means that we can never unset a pattern using the `.update()` method. :scream: